### PR TITLE
feat(limit-line-length): Show a ruler in unified diffs

### PR DESCRIPTION
**Pull request by avshalomt**
feat(limit-line-length): Show a ruler in unified diffs to mark the line length limit, configurable using a new line length limit option.

-   [x] I updated the CHANGELOG.md
-   [X] I tested the changes in this pull request myself
-   [X] I added an Option to enable / disable this feature
-   [X] I updated the README.md (without any pictures)

image

### CODE REVIEW

File: src/limit-line-length/limit-line-length.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import addStyleToPage from '../add-style'
+
+function createCssRules(lineLengthLimit, isStickyHeaderEnabled) {
+    let filler = ''
+    for (let i = 0; i < lineLengthLimit; i++) {
+        filler += '#'
+    }
+
+    let cssRules = `
+        .refract-content-container > .udiff-line:first-child .source::before,
```
Comment (avshalomt) [inline src/limit-line-length/limit-line-length.js:11]: I needed to reference only one of the `.udiff-line` elements (didn't care which one) inside `.refract-content-container`.
As far as i know this is the best way to do it (:first-of-type won't help here, nor :first-child/:last-child)

File: src/options.html [hunk 2]
```diff
@@ -134,21 +134,36 @@
             <br />
             <br />
 
+            <label>
+                <input type="checkbox" name="lineLengthLimitEnabled" />
```
Comment (avshalomt) [inline src/options.html:138]: Another possible description could be `Show a line length limit ruler at the Nth column`, what do you prefer?

**Comment by avshalomt** [comment]
@reyronald Can you take a look?

**Comment by reyronald** [comment]
Hey @avshalomt , I'm so sorry, I missed this PR originally!

Does this still work today though? Atlassian has made a lot of changes to their PR page so some of the selectors that worked in October might not today. can you re-check?

And also, can you add a screenshot to the PR description?

**Comment by avshalomt** [comment]
Hey Rey,
I'll check it again and let you know, and add the screenshot.

Should I also check on the New PR experience UI?

**Comment by avshalomt** [comment]
Working as expected (tested today) in all of these:
* PR creation page (pull-requests/new)
* PR diff page (pull-requests/<num>/.../diff)
* Commit page (/commits/hash)
* Branch compare page (/branch/<branch>?dest=<dest>)

It doesn't work when activating the new PR experience, but i didn't see any other features there.
Is it by your design?

I also added a screenshot.

**Comment by reyronald** [comment]
Hey @avshalomt, it doesn't work on the new PR experience because the entire UI was rewritten by the Bitbucket team, so every feature in the extension will need two implementations if we want it to work in both.

However, writing features for the new PR experience is considerably more challenging because the CSS class names are autogenerated with a CSS-in-JS solution, so we don't have a deterministic way of hooking up to DOM elements. If you'd like this to work in the new experience you can take a stab it if you want and let me know if you have any questions! But so far this looks good to me for the old experience.

Thanks a lot for working on it!

**Review by reyronald — APPROVED** [review]

**Pull request merged**